src/robot/dial.ts

```diff
@@ -185,3 +193,44 @@ export const createRobotClient = async (
 
   return client;
 };
+
+export const createViamTransportFactory = async (
+  serviceHost: string,
+  dialOpts: DialOptions
+): Promise<grpc.TransportFactory> => {
+  if (!dialOpts.credentials) {
+    throw new Error(`credential cannot be none`);
+  } else if (dialOpts.credentials.type === 'robot-secret') {
+    throw new Error(`credential type cannot be 'robot secret'`);
```

Comment:
Is there a place we can link to or more information to share to help consumers of this package learn the correct credential configuration?

Reply:
I don't think this is documented anywhere - let's add a doc-string to this function with these usage details.